### ISSUE TITLE
Data source blocks not working on downloaded apps

### ISSUE THREAD

**Issue by jaredgib** [thunkable/thunkable-issues#502]
I am trying to publish a project using the new data component blocks. When I download the app to my android emulator (don’t have an android device), then try to save to submit data, and save to google sheets does not work. When I try this on the live companion on my iPhone it does save.

Is anyone working with these blocks in a published app? I will update when the download comes to my iPhone.

again, this does work on live preview. not on the download to my android emulator.

this does work in:
the responsive web app preview.
live companion app on iPhone
live test in browser window

does not work on:
android emulator
the published responsive web app

### Platform
Thunkable X when using

published web apps
apps downloaded for android
apps downloaded for iOS

### Issue/Request Origin
my own issue

### Steps to Reproduce
1. link to a google sheet with data source
2. create a new entry with the live preview version
3. download app, try same data entry

### Project Link
https://x.thunkable.com/projects/5eab5a8049fc94874c55ddda/15aae766-aa26-48d5-ac67-75475a85711e/designer

### Expected Behavior
I log in. i track data. click save. I expect my data to go into my google sheet.

 this only works when i use the live preview of the app. 

### Actual Behavior
nothing happens, the app sits forever at the save point

<img width="550" alt="Screen Shot 2020-05-06 at 1 34 52 PM" src="https://user-images.githubusercontent.com/58823942/81211242-1438e100-8fa1-11ea-8825-12006263fd72.png">

this data submission occurred while using the live preview. I downloaded without making changes and had 0 luck getting the app to submit data. 
<img width="1137" alt="Screen Shot 2020-05-06 at 1 57 33 PM" src="https://user-images.githubusercontent.com/58823942/81211590-975a3700-8fa1-11ea-9574-cfe1fc461078.png">

I have tried the method block by itself, btw. Still didn’t work

**Comment by jaredgib** [comment]
I have tried the method block by itself, btw. Still didn’t work.

**Comment by StevenPurn** [comment]
Hi, sorry you're having a problem with this! Were you able to test the built app on your iOS device? Does that appear to be working as expected?

If you have a published web app would you mind sharing the link here so I can take a look to try and get an idea of what might be going wrong?

**Comment by jaredgib** [comment]
Thanks @StevenPurn 

Here’s the published web app
https://thunk.app/w/2OYC0aogK

I did download to my iPhone. It did not work unfortunately. Only on live previews did the data make it to the google sheet. the app sticks at the loading icon after clicking submit data while waiting on the data to send. I tried it without the loading icon too.

**Comment by jaredgib** [comment]
Username jgibbs@braintrustbh.com

Perfect 123456

**Comment by StevenPurn** [comment]
In my test, one of the values that the app was trying to send to the sheet was `null` which is causing an issue. I _think_ it might be the first one, `app name` for `staff value`? We should obviously be able to account for this and still have the request go through but, in the meantime, can you verify that the variable is being set correctly in your built apps and then try again?

I'll make sure we get a ticket in for fixing issues with null values. Of course it is also possible that there is another issue that just didn't come up in my test. So let's check for other issues as well once we get this fixed.

**Comment by jaredgib** [comment]
Thank you for testing. You are right about the null value. The first value was sent as null. I used the wrong variable, still though that doesn't explain the discrepancy between live test and published. But, i have fixed that and now the first value should be the email logged into the app with. 
https://thunk.app/w/2OYC0aogK

https://x.thunkable.com/projects/5eab5a8049fc94874c55ddda/15aae766-aa26-48d5-ac67-75475a85711e/blocks

https://www.loom.com/share/c7e177f7d056436fbf5dd6a79e78b175

**Comment by StevenPurn** [comment]
Great, thanks for following up. I at least see what is going wrong though I'm not sure exactly why yet. I'll look into it now and hopefully we can get this fixed soon.

**Comment by jaredgib** [comment]
Thank you. Me too. This app can’t get off the ground until I get google sheets saving working. I need to utilize the BAA we have. Airtable doesn’t offer this :(

Thank you 🙏

**Comment by jaredgib** [comment]
It works I think. I’ve tested this on iOS. 
Idk what you guys do but it’s like magic!!! 

Thank you!

**Comment by StevenPurn** [comment]
So after you fixed the issue with the null variable it worked on the built iOS app? Have you had a chance to test Android? I've got a fix for the published web app that I'm doing some testing on now and hoping to get it out in the next couple days.

**Comment by jaredgib** [comment]
It is bizarre. The app I downloaded last night, didn't work. Then I woke up and tried it one more time before following up, and it worked. IDK, man. Those app ferries were at work. Or maybe I was so tired I didn't realize that I had changed it already. But it's working on iOS. I will test on android in an hour or so

Thanks again!

**Comment by jaredgib** [comment]
Hey @StevenPurn, this is working on iOS and android both. I am still unable to send data to the google sheet using the web app. To my knowledge, there are no 'null' values being sent. Am I doing something wrong here or is this a bug to be worked out?

Here is the app.
https://thunk.app/w/2OYC0aogK

and the project
https://x.thunkable.com/projects/5eab5a8049fc94874c55ddda/564bf3e0-f21f-42f9-a439-1d9596f416c8/blocks

**Comment by StevenPurn** [comment]
No, you aren't doing anything wrong, there is an issue preventing the web app from working with data sources at the moment. We have a fix that has been approved and will be included in our next release. It will likely be live early next week.  Hopefully Tuesday at the latest. Sorry for the delay! Thanks for confirming Android and iOS are working.

**Comment by jaredgib** [comment]
THanks @StevenPurn! I am looking forward to this. I just received a denial from Apple for my this app as it is designed for my workplace only. I will be working on making it accessible for anyone that wants to send a 'healthy pre-work screening' to their HR dept and have it back to apple by the end of tomorrow, but I am hoping to be able to use this web app for my work specifically!

Thank you again!

**Comment by StevenPurn** [comment]
You should not need to make any changes for the fix to be applied to your project once it is deployed. I will check in with the team and let you know as soon as I have an idea as to when that will be.

**Comment by jaredgib** [comment]
That's great! I know I don't speak for the community expressly, but I believe a lot of us are noticing an increased amount of communication from you guys lately! It's really awesome and is just one reason I continue to love this platform and hope it continues to grow, mature and excel! We all appreciate it! (like that recently shared google sheet)

**Comment by jaredgib** [comment]
Hello @stevenpurn! I’m wondering if you have any updates on this? I checked yesterday and was unable to save the data still. 

Would it be helpful if I make a second project to help root out the issue?

**Comment by StevenPurn** [comment]
Sorry, I expected it to be released by now. No, no further action is required on your end. We have a fix in, just waiting on it to be deployed to our production servers. It looks like we are getting ready to release it soon though. There might have been another issue which was slowing down the process. I'll push to try and get it out today if we can but I can't make any promises unfortunately.

**Comment by StevenPurn** [comment]
It looks like we had things ready to go out but there was a problem with our iOS app review process. We typically like to release to all platforms simultaneously to ensure there aren't any issues. We got the approval from Apple today though so I expect everything will be deployed today at some point. Sorry again for the delay!

**Comment by jaredgib** [comment]
It's working perfectly! Thank you guys so much!!!!

Tonight I have published (and been paid for) two web apps! This fix made 1 of those possible!

**Comment by StevenPurn** [comment]
Glad to hear! Sorry that it wasn't working as intended earlier. 

We appreciate your help in both identifying and resolving these issues! 😄
